src/vs/workbench/api/common/extHostTypes.ts

```diff
@@ -3075,18 +3075,80 @@ export class FunctionBreakpoint extends Breakpoint {
 
 @es5ClassCompat
 export class DataBreakpoint extends Breakpoint {
-	readonly label: string;
-	readonly dataId: string;
-	readonly canPersist: boolean;
+	private _resolution?: vscode.DataBreakpointResolution;
 
-	constructor(label: string, dataId: string, canPersist: boolean, enabled?: boolean, condition?: string, hitCondition?: string, logMessage?: string, mode?: string) {
+	readonly source: vscode.DataBreakpointSource;
+	readonly accessType: vscode.DataBreakpointAccessType;
+
+	constructor(source: vscode.DataBreakpointSource, accessType: vscode.DataBreakpointAccessType, enabled?: boolean, condition?: string, hitCondition?: string, logMessage?: string, mode?: string) {
 		super(enabled, condition, hitCondition, logMessage, mode);
-		if (!dataId) {
-			throw illegalArgument('dataId');
+		this.source = source;
+		this.accessType = accessType;
+		if (source instanceof ResolvedDataBreakpointSource) {
+			this.set({ ...this.source });
 		}
-		this.label = label;
-		this.dataId = dataId;
-		this.canPersist = canPersist;
+	}
+
+	protected async info(session: vscode.DebugSession): Promise<vscode.DataBreakpointResolution> {
+		// ResolvedDataBreakpointSource is already handled in the constructor
+		if (this.source instanceof AddressDataBreakpointSource) {
+			return session.customRequest('dataBreakpointInfo', { name: this.source.address, bytes: this.source.bytes, asAddress: true }) as Promise<vscode.DataBreakpointResolution>;
+		}
+		if (this.source instanceof ExpressionDataBreakpointSource) {
+			return session.customRequest('dataBreakpointInfo', { name: this.source.expression }) as Promise<vscode.DataBreakpointResolution>;
+		}
+		if (this.source instanceof VariableScopedDataBreakpointSource) {
+			return session.customRequest('dataBreakpointInfo', { name: this.source.variable, variablesReference: this.source.variablesReference }) as Promise<vscode.DataBreakpointResolution>;
+		}
+		if (this.source instanceof FrameScopedDataBreakpointSource) {
+			return session.customRequest('dataBreakpointInfo', { name: this.source.expression, frameId: this.source.frameId }) as Promise<vscode.DataBreakpointResolution>;
+		}
+		throw Error('Unknown data breakpoint source');
+	}
+
+	set(resolution: vscode.DataBreakpointResolution): this {
+		if (!this.resolution) {
+			this._resolution = resolution;
+		}
+		return this;
+	}
+
+	async resolve(session: vscode.DebugSession): Promise<this> {
+		return this.set(await this.info(session));
+	}
+
+	get resolution(): vscode.DataBreakpointResolution | undefined {
+		return this._resolution;
+	}
+}
+
+@es5ClassCompat
+export class ResolvedDataBreakpointSource {
+	constructor(public readonly dataId: string, public readonly canPersist: boolean = false, public readonly accessTypes?: vscode.DataBreakpointAccessType[], public readonly description: string = `DataId '${dataId}'`) {
+	}
+}
+
+@es5ClassCompat
```

Comment:
es5ClassCompat is not needed on new classes

Reply:
Fixed.